packages/client/src/models/parent.js

```diff
@@ -20,55 +21,112 @@ const ParentConverter = {
         return {
             name: parent.name,
             email: parent.email,
+            phone: parent.phone,
+            pronouns: parent.pronouns,
+            avatar: parent.avatar,
             timezone: parent.timezone,
-            mentees: mentees
+            students: parent.students
         };
     },
 
     fromFirestore: (snapshot, options) => {
         const data = snapshot.data(options);
 
-        const mentees = [];
+        const students = [];
 
-        data.mentees.forEach(s => {
-            mentees.push(new Mentee(s.name, s.email, s.grade, s.subjects))
+        data.students.forEach(s => {
+            students.push(new Student(s.name, s.email, s.grade, s.subjects))
         });
 
         return new Parent(
             data.name,
             data.email,
+            data.phone,
+            data.pronouns,
+            data.avatar,
             data.timezone,
-            mentees
+            data.number_requests,
+            students
         );
     }
 }
 
+const phoneRegex = RegExp(
+    /^\(?([0-9]{3})\)?[-. ]?([0-9]{3})[-. ]?([0-9]{4})$/
+);
+
+const parentSchema = yup.object().shape({
+    email: yup
+        .string()
+        .email()
+        .required('Email Required'),
+    name: yup
+        .string()
+        .required('Name Required'),
+    timezone: yup
+        .string()
+        .required('Timezone Required'),
+    phone: yup
+        .string()
+        .matches(phoneRegex, 'Phone number is not valid'),
```

Comment:
Is is possible in Yup to require either an email or a phone number but not both simultaneously?

Reply:
Would this work? 

```js
yup.string().test(function (value) {
  const { email } = this.parent;
  if (!email) return value != null
  return true
})
```

Reply:
I don’t think so. `this` is not available in the Yup schema.

Reply:
then can we just use `parent`?